Tutorial: gMCS

lvalcarcelg opened the pull request:
Tutorial for the calculation of genetic Minimal Cut Sets.

**I hereby confirm that I have:**

- [X] Tested my code on my own machine
- [X] Followed the guidelines in the [Contributing Guide](https://opencobra.github.io/cobratoolbox/docs/contributing.html)
- [X] Selected `develop` as a target branch (top left drop-down menu)

*(Note: You may replace [] with [X] to check the box)*

laurentheirendt commented:
@lvalcarcelg, thanks for this 👍 

We are generating the PDF and the HTML automatically. May you remove them? Thanks!

laurentheirendt commented:
@tpfau, may you check the part here with I/O? Thanks 👍

tpfau commented on tutorials/analysis/gMCS/tutorial_gMCS.m:
```diff
@@ -0,0 +1,227 @@
+% First, we are going to load the MAT-file which contains the metabolic network 
+% in the toy example.
+%%
+load('gMCStoyExample.mat');
```
please use readCbModel instead of load.
This is to ensure that the models used are definitely compatible with the toolbox, and thus that the functions called are. 
Please also make sure, that this is also updated in the mlx file.

lvalcarcelg commented:
I have changed all files to contain readCbModel. In addition, I have deleted the MAT file which contains the toy model from the test folder. 
I prefer to have the model in the tutorial folder, for easier understanding.

tpfau commented on tutorials/analysis/gMCS/tutorial_gMCS.m:
```diff
@@ -0,0 +1,227 @@
+% First, we are going to load the metabolic model.
+
+global CBTDIR
+load([CBTDIR filesep 'test' filesep 'models' filesep 'mat' filesep 'Recon2.v04.mat']);
```
Sorry, I didn't notice this one before.
I'd suggest to either use `getDistributedModel('Recon2.v04.mat')` here, which is more stable, as it will automatically update the path if that ever changes, or use `readCbModel`, if you want to be clearer in indicating the source of the model. (Again also update the mlx file)

lvalcarcelg replied:
Thanks for the comment. I am also working in a little mistake in the test file. Now is working in my PC, so I think it will work in Jenkins.

tpfau commented:
Just wondering, is this "NaN" entry necessary? Or could the output just be truncated? 
Essentially, the size of the resulting vector should indicate the number of possibilities if the maximal number is larger than the number of actual existing strategies. 
I find this NaN rather unintuitive and don't really see a reason for it.

tpfau commented:
One other thing I noticed:
it might be good to initialize the optional_inputs struct before assigning values.
Mainly to avoid other settings being used that stem from a different run and where not removed.

iaemparanza commented:
Thanks @tpfau for the interesting observations! Initially, we thought it would be easier for the user to see that no more feasible solutions exist with a NaN. However, I think you are right. We are now correcting the functions to truncate the solutions when no more feasible ones exist.
In addition, we are also working on the tutorial to initialize the optional_inputs variables in order to avoid any problems.
Thanks very much for you comments!
